static/js/settings_ui.js

```diff
@@ -66,6 +66,64 @@ exports.do_settings_change = function (request_method, url, data, status_element
     });
 };
 
+// Generic function for informing users about changes to the settings
+// UI. Uses the save_discard_button_widget used in other parts of the
+// application. Works by hiding the discard_button so that only the
+// save button is visible, and whose text is modified to display the
+// current status.
```

Comment:
Can you talk about what the benefit of this over `do_setting_change` is ?  

I think the use case is for models or other widgets that require a "Save" button for UI reasons, but the documentation doesn't say that, so it's hard for me to be sure.  

As a general rule, explaining the purpose of things is a much more important use case for comments than explaining how something works -- one can see how it works by reading the code itself.

Reply:
Though see the comment below; it's probably the case that we don't need this as an independent function.